src/olympia/addons/models.py

```diff
@@ -487,12 +487,15 @@ def delete(self, msg='', reason=''):
     @classmethod
     def initialize_addon_from_upload(cls, data, is_listed=True):
         fields = cls._meta.get_all_field_names()
-        # Reclaim GUID from deleted add-on.
-        try:
-            old_guid_addon = Addon.unfiltered.get(guid=data['guid'])
-            old_guid_addon.update(guid=None)
-        except ObjectDoesNotExist:
-            old_guid_addon = None
+        guid = data.get('guid')
```

Comment:
Well, `guid` is a `CharField` but doesn't have `blank=True` (default for `CharField` is `False`) and is nullable too. The `blank` thing is only relevant in forms and not in the database though so we might have both in the database.

But since it's explicitly nullable I thought we were only using null here for empty-values.